### DISCUSSION TITLE
Import from CSV will hang on processing, missing mapping property's?

### DISCUSSION THREAD

**Discussion by sleeman1** [wgu-opensource/osmt#147] [Asking for Help] [answered]
We have a demo csv of skills from WGU to test import on a demo QuickStart build
[Software Product Management.csv](https://github.com/wgu-opensource/osmt/files/7989656/Software.Product.Management.csv)
If we just "Next" through the process the import will hang.
![hanging import](https://user-images.githubusercontent.com/64618524/152228250-e4c29fee-c924-4c22-ade1-0a7cada80da0.jpg)

It appears be a column mapping issue, mapping two columns to the same record or undefined column names? When we have removed a columns from the CSV  that don't have a mapped property it appears to work importing 98 RDS.  You can see in the image  below we have missing mappings.  for data like "Major Group (BLS)" Broad Occupation (BLS)" or "Job Role (O*NET)  is it expected we select "Do not import" for this demo CSF data?
![image](https://user-images.githubusercontent.com/64618524/152230925-37726c63-0e5c-4628-bf6b-6f5bfb34b385.png)

In the Readme I see for the Dev server you need to import BLS and O*Net codes. Could it be that the I am assuming wrong  and QuickStart demo does not not have the code data?

Thoughts?

Thanks for the advice!
Chad

**Comment by JohnKallies** [comment]
Hi Chad.

Quickstart's docker_entrypoint.sh script imports the default BLS and O*NET codes, from csv files in the `import` directory in the project root. You _can_ override that behavior with a couple flags in the `osmt-quickstart.env` file (REINDEX_ELASTICSEARCH and SKIP_METADATA_IMPORT), so it's worth double checking that imports and reindexing are still enabled (compare those env vars to `osmt-quickstart.env.example` to see the defaults).

Otherwise, a Quickstart _should_ have those job codes imported.
* You should be able to connect to the Quickstart docker-compose stack's MySQL instance and verify it (`select count(*) from JobCode;` should return 2425)
* You can also create a new RSD, and start typing in the "Occupations" field (those job codes help auto-complete that field).

Regarding column mapping, I am not an expert on using this feature. I have one file that I test with on a regular basis, and I haven't explored what this feature can do / what its limitation are.
* @sleeman1, would you please confirm that the job codes are imported, using either method above?
* I'll share your question with some people who have deeper experience with using OSMT, and see what I can turn up.

Thank you,
John

**Comment by JohnKallies** [comment, marked as answer]
Hi Chad @sleeman1,

I got some help from a skills person within WGU. She was successful in importing your file after mappings all of these fields to "Occupations"
* Occupation Major Groups
* Occupation Minor Groups
* Borad Occupations
* Detailed Occupations

Please let me know if this resolves it for you? Thank you.

**Reply by JohnKallies** [reply]
I just now see that this is covered in the How-To section on https://osmt.io, in the "Add Skills to OSMT: How to Batch Import" section.

**Reply by sleeman1** [reply]
@JohnKallies  what did the skills person map "Title for Alignment" and "Alignment" to? We have tried the Alignment Names and Alignment URLs,  and Do not Import with no luck.

We appreciate the help!

**Reply by JohnKallies** [reply]
"If he was using the CSV file you shared with me, it's missing a column for import. The template required for batch import has an additional job code level: Occupation Minor Group. It could be this misalignment that is causing the error. I haven't tried it myself because I wasn't able to batch import using their CSV...I copied the data to the template provided in OSMT."
